docs/security.md

```diff
@@ -207,6 +207,38 @@ For this feature to work, additional configuration settings are needed in Hadoop
 </property>
 ```
 
+####Application Root Directory under HDFS
+
+A running Apex application uses a 'root' directory under HDFS where runtime artifacts are saved or read from. For example, with default configuration an Apex application 
+would use the HDFS path `/user/dtadmin/datatorrent` as the application root directory. If the application's Hadoop assigned application ID is `application_1487803614053_10222` 
+then the full HDFS path for saving and reading application package contents,  events, checkpoint and recovery data is 
+`/user/dtadmin/datatorrent/apps/application_1487803614053_10222`.
+
+This root directory determination involves using the 'current' user, and in case of impersonation Apex treats either the impersonating user or the impersonated 
+user as the 'current' user depending on the value of the configuration propery `dt.authentication.impersonation.path.enable`. You should set its value to `true` to 
```

Comment:
@sanjaypujare dt.authentication.impersonation.path.enable was introduced since last release, please change this to apex. prefix. Perhaps you can do that as part of APEXCORE-755, but before release 3.7

Also, there are a few pre-existing incorrectly formatted headers in this doc that would be good to fix (like ####Hadoop configuration approach and following). Working on existing files is an opportunity to cleanup such trivial issues.